subsys/net/lib/ftp_client/src/ftp_client.c

```diff
@@ -75,46 +79,50 @@ static int parse_return_code(const uint8_t *message, int success_code)
 static int establish_data_channel(const char *pasv_msg)
 {
 	int ret;
-	char tmp[16];
+	char tmp[8];
 	char *tmp1, *tmp2;
 	uint16_t data_port;
 
 	/* Parse Server port from passive message
-	 * e.g. "227 Entering Passive Mode (90,130,70,73,86,111)"
+	 * e.g. "227 Entering Passive Mode (90,130,70,73,86,111)" in case of IPv4
+	 * e.g. "227 Entering Passive Mode (0,0,0,0,97,78)" in case of IPv6
+	 * NOTE assume no IP address change from the Control channel
 	 */
-	tmp1 = strstr(pasv_msg, "(");
-	tmp1++;
-	tmp2 = strstr(tmp1, ",");
-	tmp2++;
-	tmp1 = strstr(tmp2, ",");
-	tmp1++;
-	tmp2 = strstr(tmp1, ",");
-	tmp2++;
-	tmp1 = strstr(tmp2, ",");
-	tmp1++;
-	tmp2 = strstr(tmp1, ",");
-	memset(tmp, 0x00, 16);
-	strncpy(tmp, (const char *)tmp1, (size_t)(tmp2 - tmp1));
-	data_port = atoi(tmp) << 8;
-	tmp2++;
-	tmp1 = strstr(tmp2, ")");
-	memset(tmp, 0x00, 16);
-	strncpy(tmp, (const char *)tmp2, (size_t)(tmp1 - tmp2));
-	data_port += atoi(tmp);
-
-	/* Establish the second connect for data */
+	tmp1 = strrchr(pasv_msg, ')');
+	if (tmp1 == NULL) {
+		return -EINVAL;
+	}
+	tmp2 = strrchr(pasv_msg, ',');
+	if (tmp2 == NULL) {
+		return -EINVAL;
+	}
+	memset(tmp, 0x00, 8);
+	strncpy(tmp, (const char *)(tmp2 + 1), (size_t)(tmp1 - tmp2 - 1));
+	data_port = atoi(tmp);
+	tmp1 = tmp2 - 1;
+	while (isdigit((int)(*tmp1))) {
+		tmp1--;
+	}
+	if (*tmp1 != ',') {
+		return -EINVAL;
+	}
+	memset(tmp, 0x00, 8);
```

Comment:
nit: Use `sizeof(tmp)` instead of magic number `8`.

Reply:
Updated, thanks